app/javascript/admin/controllers/config_controller.js

```diff
@@ -78,30 +98,156 @@ export default class ConfigController extends Controller {
+    const provider = event.target.dataset.authProvider;
+    const official_provider = event.target.dataset.authProviderOfficial;
+    this.configModalAnchorTarget.innerHTML = adminModal(
+      this.authProviderModalTitle(official_provider),
+      this.authProviderModalBody(official_provider),
+      'Confirm disable',
+      'disableAuthProviderFromModal',
+      'Cancel',
+      'closeAdminConfigModal',
+      'auth-provider',
+      provider,
+    );
+    this.positionModalOnPage();
   }
+
+  disableAuthProviderFromModal(event) {
+    event.preventDefault();
+    const provider = event.target.dataset.authProvider;
+    const authEnableButton = document.querySelector(
+      `[data-auth-provider-enable="${provider}"]`,
+    );
+    const enabledIndicator = document.querySelector(
+      `#${provider}-enabled-indicator`,
+    );
+    authEnableButton.setAttribute('data-enable-auth', 'false');
+    this.listAuthToBeEnabled(event);
+    this.checkForAndGuardSoleAuthProvider();
+    this.showHideAuthEnabledIndicator(enabledIndicator, 'hide');
+    this.hideAuthProviderSettings(event);
+    this.closeAdminConfigModal(event);
+  }
+
+  checkForAndGuardSoleAuthProvider() {
+    if (
+      document.querySelectorAll('[data-enable-auth="true"]').length === 1 &&
+      document
+        .querySelector('#email-auth-enable-edit-btn')
+        .getAttribute('data-button-text') === 'enable'
+    ) {
+      const targetAuthDisableBtn = document.querySelector(
+        '[data-enable-auth="true"]',
+      );
+      targetAuthDisableBtn.parentElement.classList.add('crayons-tooltip');
+      targetAuthDisableBtn.parentElement.setAttribute(
+        'data-tooltip',
+        'To edit this, you must first enable Email Address as a registration option',
+      );
+      targetAuthDisableBtn.setAttribute('disabled', true);
+    }
+  }
+
+  hideAuthProviderSettings(event) {
+    event.preventDefault();
+    const provider = event.target.dataset.authProvider;
+    document
+      .querySelector(`#${provider}-auth-settings`)
+      .classList.add('hidden');
+    document.querySelector(`#${provider}-auth-btn`).classList.remove('hidden');
+  }
+
+  listAuthToBeEnabled() {
+    const enabledProviderArray = [];
+    document
+      .querySelectorAll('[data-enable-auth="true"]')
+      .forEach((provider) => {
+        enabledProviderArray.push(provider.dataset.authProviderEnable);
+      });
+    document.querySelector(
+      '#auth_providers_to_enable',
+    ).value = enabledProviderArray;
+  }
+
+  disableAuthenticationOptions() {
+    document.querySelector('#auth_providers_to_enable').value = '';
+  }
+
+  showHideAuthEnabledIndicator(indicatorDiv, action) {
+    if (action === 'show') {
+      indicatorDiv.classList.add('flex', 'items-center');
+      indicatorDiv.classList.remove('hidden');
+    } else if (action === 'hide') {
+      indicatorDiv.classList.remove('flex', 'items-center');
+      indicatorDiv.classList.add('hidden');
+    }
+  }
```

Comment:
It would be ideal if we didn't have to pass in `'show'` and `'hide'` here to figure out what class should be rendered on the div. I haven't tried it locally yet, but I _think_ it should be possible to remove the conditional entirely and toggle the class, instead. As long as the div starts out with a default class, it _should_ work...I'm imagining something like this:
```javascript
toggleAuthEnabledIndicator(indicatorDiv) {
  indicatorDiv.classList.toggle('visible');
}
```

I think you can port more of this over to CSS and slim down JavaScript. This means that you can't use the generated utility classes, but IMO, that's okay here.

We could do this with CSS in `admin.scss` by adding a class to the `indicatorDiv` element(s):
```
.your-class-here {
  display: none; // The default visibility is hidden
}

.visible { // You can toggle this class and it should theoretically override the default styles
  display: flex;
  align-items: center;
}
```

Ideally, you wouldn't to check if we're trying to hide/show, you'd just toggle visibility on/off by calling the function at the right time. [Here's a short tutorial](https://www.digitalocean.com/community/tutorials/js-classlist) that might be helpful if you want to investigate it further!

Let me know what you think :)

Reply:
@nickytonline and @ludwiczakpawel : What's your point-of-view here on whether we accomplish a toggle using Javascript OR CSS? I don't think we have this "best practice" defined in our docs yet, and would love to hear your thoughts.

Reply:
@lisasy Something I want to add here is that sometimes it's not always possible to remove _all_ the JS, but the more that we can lean into CSS, the better.

CSS renders/parses faster than JS. CSS fails much more elegantly than JavaScript (CSS is more forgiving, whereas JS is more error prone and you have to assume or _check_ that an element exists in the DOM before you can do anything to it). Also, when used correctly, CSS is easier to maintain and debug (I can't say the same about JS 😅).

I can't remember where I originally learned about this years ago, but [here](https://gomakethings.com/when-to-use-css-vs.-javascript) is a relevant blog post.

Reply:
Done! @vaidehijoshi